api/types/ui_config.go

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package types
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// UIConfig defines configuration for the web UI served
+// by the proxy service. This is a configuration resource,
+// never create more than one instance of it.
+type UIConfig interface {
+	Resource
+
+	// GetScript returns the contents of the installer script
+	GetScrollbackLines() int32
+	// SetScript sets the installer script
+	SetScrollbackLines(int32)
```

Comment:
the `number` type in js is 64bit so I think this should probably be `int64`

Reply:
Number in JavaScript is a double precision floating point number. You can't have a fractional number of lines.

The maximum int32 is 2 billion - given that the current max is 1000 lines, I think 2 billion is plenty.

Reply:
😆 Good point